### PR TITLE
EOS-26739:Performance Stats: Authenticate client consuming metrics based on conf-store key provided.

### PR DESCRIPTION
**Pull request by Pranali04796**
# Problem Statement
- Authenticate client consuming metrics based on conf-store key provided.
- [EOS-26739](https://jts.seagate.com/browse/EOS-26739)

# Design
- Introduced a new decorator which can help change behavior of existing api's based on conf-key value.

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with custom image

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: csm/conf/etc/csm/csm.conf [hunk 1]
```diff
@@ -76,7 +76,7 @@ STATS:
         ssl_check: "false"
         interval: "10" # Flush interval in sec (Rate at which stats is stored)
         offset: "20" # offset in sec
-
+    public_auth: "true"
```
Comment (pawankumarsrivastava) [inline csm/conf/etc/csm/csm.conf:79]: Change the key from public_auth to auth (enable / disable)

File: csm/core/agent/api.py [hunk 2]
```diff
@@ -312,11 +317,26 @@ async def _validate_bearer(login_service: LoginService, session_id: str):
...
+
     @staticmethod
     @web.middleware
     async def session_middleware(request, handler):
         session = None
         is_public = await CsmRestApi._is_public(request)
+        is_hybrid = await CsmRestApi._is_hybrid(request)
+        if is_hybrid:
+            conf_key = CsmRestApi._retrieve_config(request)
+            if conf_key == "true" or conf_key == "True":
```
Comment (pawankumarsrivastava) [inline csm/core/agent/api.py:336]: Change the logic based on new key mentioned above

File: csm/core/blogic/const.py [hunk 3]
```diff
@@ -915,4 +915,5 @@
 AGENT_ENDPOINTS = 'CSM_SERVICE>CSM_AGENT>endpoints'
 AGENT_HOST = 'CSM_SERVICE>CSM_AGENT>host'
 AGENT_PORT = 'CSM_SERVICE>CSM_AGENT>port'
-AGENT_BASE_URL = 'CSM_SERVICE>CSM_AGENT>base_url'
\ No newline at end of file
+AGENT_BASE_URL = 'CSM_SERVICE>CSM_AGENT>base_url'
+PUBLIC_AUTH = 'STATS>public_auth'
```
Comment (pawankumarsrivastava) [inline csm/core/blogic/const.py:919]: Change this too

File: csm/core/controllers/stats.py [hunk 4]
```diff
@@ -112,3 +112,14 @@ async def get(self):
...
+
+@CsmAuth.hybrid
+@CsmView._app_routes.view("/api/v2/metrics/stats/perf")
+class MetricsView(CsmView):
+    def __init__(self, request):
+        super().__init__(request)
+        self._service = self.request.app["stat_service"]
+    @CsmAuth.permissions({Resource.STATS: {Action.LIST}})
+    async def get(self):
+        Log.debug(f"Handling Metrics")
```
Comment (pawankumarsrivastava) [inline csm/core/controllers/stats.py:124]: f"Handling Metrics" ---> "Handling get request for performance stats"

**Review by pawankumarsrivastava — CHANGES_REQUESTED** [review]

File: csm/conf/etc/csm/csm.conf (hunk 1 above)
Comment (Pranali04796) [reply csm/conf/etc/csm/csm.conf]: Updated key.

File: csm/core/agent/api.py (hunk 2 above)
Comment (Pranali04796) [reply csm/core/agent/api.py]: resolved

File: csm/core/blogic/const.py (hunk 3 above)
Comment (Pranali04796) [reply csm/core/blogic/const.py]: resolved by updating key

File: csm/core/controllers/stats.py (hunk 4 above)
Comment (Pranali04796) [reply csm/core/controllers/stats.py]: resolved

**Review by pawankumarsrivastava — APPROVED** [review]

**Pull request merged**